core/src/main/scala/com/spotify/featran/FeatureSpec.scala

```diff
@@ -86,6 +86,21 @@ class FeatureSpec[T] private[featran] (private[featran] val features: Array[Feat
     new FeatureSpec[T](this.features, this.crossings + (k -> f))
   }
 
+  /**
+   * Extend this Spec with another Spec but allowing for a function to be called first.
+   * Useful for an existing Spec on a subobject.
+   * @param f Function to apply before calling the extending spec.
+   * @param spec Spec to apply.
+   * @tparam S Type of the extension spec
+   */
+  def extend[S](f: T => S)(spec: FeatureSpec[S]): FeatureSpec[T] = {
```

Comment:
This looks like a `specA.map[B](f: A => B)` followed by a `MultiFeatureSpec(specA, specB)`? Is that a better modular approach?

Reply:
Not exactly a multispec.  I don't really want a separate list of vectors on put here instead I want to reuse some specs but still get out a single vector as normal.  Could be more difficult to deal with the multiple vectors and names in that case.

So if you have a feature objects like 

```
case class House(sqft: Double, color: String)
case class HousePrice(house: House, price: Float)
case class HouseSold(house: House, sold: Boolean)
```

It would be nice to write one House spec and use for both price and sold without needing to rewrite code and worry about multiple splits.

Reply:
Oh I get now. What about this?
`def compose(spec: FeatureSpec[S])(f: T => S)`

`compose` feels more natural (using another spec as part of this), and `f` as 2nd arg so it's easy to write multi-line fn.

Reply:
Makes sense, let me change it to that.  I was debating between compose and extend.